File: 프로그래머스/1/12926. 시저 암호/시저 암호.cs
public class Solution {
    public string solution(string s, int n) {
        char[] sArray = s.ToCharArray();
        
        for(int i = 0; i < sArray.Length; i++) {
            if(sArray[i] == ' ') continue;
            
            int a = (sArray[i] >= 'A' && sArray[i] <= 'Z') ? (sArray[i] + n > 'Z' ? sArray[i] - ('Z' - 'A' + 1) : sArray[i]) : (sArray[i] + n > 'z' ? sArray[i] - ('z' - 'a' + 1) : sArray[i]);
            
            sArray[i] = (char) (a + n);
        }
        
        return new string(sArray);
    }
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    
    int a = 0;
    for(int i = 0; i < s.length(); i++) {
        if(s[i] == ' ') {
            answer += " ";
        } else {
            int start = isupper(s[i]) ? 'A' : 'a';
            answer += start + (s[i]-start + n)%26;
        }
    }
    return answer;
}